proto/cacheclient.proto

```diff
@@ -435,6 +437,47 @@ message _ListLengthResponse {
   }
 }
 
+message _ListTrimRequest {
+  bytes list_name = 1;
+  optional sint32 start = 2;
+  optional sint32 end = 3;
+  optional sint32 step = 4;
```

Comment:
What do you envision step doing in this API? I believe we are looking for an API which drops elements before start and after end, leaving any elements in between - I'm not sure how a step factors into this?